apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java

```diff
@@ -124,29 +129,32 @@ public boolean performSmokeTest(RdaSink<TMessage, TClaim> sink) throws Exception
         sink.readMaxExistingSequenceNumber().orElse(MIN_SEQUENCE_NUM);
     log.info("smoke test: startingSequenceNumber={}", startingSequenceNumber);
 
-    // Call the RDA API version service to confirm the API is accessible.
-    final String apiVersion = caller.callVersionService(channel, callOptionsFactory.get());
-    log.info("smoke test: apiVersion={}", apiVersion);
+    if (serverType == RdaSourceConfig.ServerType.Remote) {
+      // Call the RDA API version service to confirm the API is accessible.
+      final String apiVersion = caller.callVersionService(channel, callOptionsFactory.get());
+      log.info("smoke test: apiVersion={}", apiVersion);
 
-    // Doesn't use startingSequenceNumber because we should not block waiting for new data.
-    final GrpcResponseStream<TMessage> responseStream =
-        caller.callService(channel, callOptionsFactory.get(), MIN_SEQUENCE_NUM);
-    try {
-      for (int i = 1; i <= 3 && responseStream.hasNext(); ++i) {
-        final TMessage message = responseStream.next();
-        log.info(
-            "smoke test: successfully downloaded claim: seq={}",
-            sink.getSequenceNumberForObject(message));
-      }
-      successful = true;
-    } finally {
-      // be a nice client that lets the server know when we are leaving before the stream is done
-      if (successful && responseStream.hasNext()) {
-        responseStream.cancelStream("smoke test: finished");
+      // Doesn't use startingSequenceNumber because we should not block waiting for new data.
+      final GrpcResponseStream<TMessage> responseStream =
+          caller.callService(channel, callOptionsFactory.get(), MIN_SEQUENCE_NUM);
+      boolean successful = false;
+      try {
+        for (int i = 1; i <= 3 && responseStream.hasNext(); ++i) {
+          final TMessage message = responseStream.next();
+          log.info(
+              "smoke test: successfully downloaded claim: seq={}",
+              sink.getSequenceNumberForObject(message));
+        }
+        successful = true;
+      } finally {
+        // be a nice client that lets the server know when we are leaving before the stream is done
+        if (successful && responseStream.hasNext()) {
+          responseStream.cancelStream("smoke test: finished");
+        }
       }
     }
 
-    return successful;
+    return true;
```

Comment:
If successful is false, why do we no longer need to pass it up the chain?

Reply:
Makes sense how it works, though I do agree it might be a bit confusing with the given language.  Maybe instead of `success` it is `exceptionThrown` so as to indicate it's merely a flag for the `finally` statement, and not directly related to the response of the test method?

Reply:
Looking at it with fresh eyes I realized that the try/finally/successful thing was unnecessary since the cancel was only ever done if no exception was thrown.